### PR TITLE
[feature] Compound Filters (input & date)

### PR DESCRIPTION
**Pull request by ghiscoding**

### CODE REVIEW

**Comment by ghiscoding** [comment]
@jmzagorski 
If you have a chance to review by Monday night, else I will merge it anyway. Also note that there's more than just Compound Filters in the PR, I found a few issues with Pagination and Filters at work and a few of these fixes are within the PR.

**Comment by jmzagorski** [comment]
I will review it today. Sorry I have been MIA on some of the tasks I said I was going to do this week. I had to rewrite many of my server Apis.

**Comment by ghiscoding** [comment]
No worries, I assumed you were busy, hence if you can't do it then I will merge it. No big issues.

File: aurelia-slickgrid/src/aurelia-slickgrid/filters/compoundInputFilter.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      }
+    }
+  }
+
+  /**
+   * destroy the filter
+   */
+  destroy() {
+    if (this.$filterElm) {
+      this.$filterElm.off('keyup').remove();
```
Comment (jmzagorski) [inline aurelia-slickgrid/src/aurelia-slickgrid/filters/compoundInputFilter.ts]: do you need a `this.$selectOperatorElm.off('change')` here?

Comment (ghiscoding) [reply aurelia-slickgrid/src/aurelia-slickgrid/filters/compoundInputFilter.ts]: oh right, I could do that one too.

**Comment by ghiscoding** [comment]
Not always equals, it depends on the type. If it's a Type String it should be Contains for the rest it will be Equals

**Comment by jmzagorski** [comment]
oh okay I see, I did not test that out in example 5 on the `Name` column

**Comment by ghiscoding** [comment]
Hmm I might have forgot to test that portion in a regular grid (without backend service), we should probably test it out in Example 4 or any other. I tested it in OData/GraphQL Backend Services and I have code to handle that. I think the regular grid are handled directly by their `filterConditions` but I can't remember totally

**Comment by ghiscoding** [comment]
It's actually a good thing you talked about this since yes I forgot to test it with a regular grid and it is in fact not working correctly. I just fixed it in my Angular-Slickgrid repo with this [commit](https://github.com/ghiscoding/Angular-Slickgrid/commit/22f960d8d144baa34df31a6313024fe0e4e0c1ab). I will apply it to the Aurelia repo tonight. 

Or perhaps if you have time, I wouldn't mind if you do the changes that you mentioned and the fix for regular grid that I just talked about (on my branch). Or I can try to do them quickly during my lunch time in about 2 hours.

**Comment by jmzagorski** [comment]
I can look over your angular commit and see how to fix it.

**Comment by ghiscoding** [comment]
I assume that you are able to do the changes on the same branch right? It should be `features/compound-filters` if I remember correctly

**Comment by jmzagorski** [comment]
yup, I have it pulled down on my machine

**Comment by jmzagorski** [comment]
@ghiscoding  
the same exact change you made in Angular works in Aurelia too. You should now be able to change local filters in example 4 and the data should filter as you expect.

**Comment by ghiscoding** [comment]
Yup I try to re-use as much code as possible between the 2 repos 😄 
I'll push a release tonight then, thx for the update

**Pull request merged**